gemrb/core/DisplayMessage.cpp

```diff
@@ -187,30 +190,65 @@ void DisplayMessage::DisplayString(String text, const Color &color, Scriptable *
 	}
 }
 
+std::vector<Color> DisplayMessage::GetAllColors() const
+{
+	std::vector<Color> auxiliaryColors;
+	if (auxiliaryColors.empty()) {
+		AutoTable colorTable = gamedata->LoadTable("colors", true);
+		assert(colorTable);
+		for (TableMgr::index_t r = 0; r < colorTable->GetRowCount(); r++) {
+			ieDword c = colorTable->QueryFieldUnsigned<ieDword>(r, 0);
+			auxiliaryColors.push_back(Color(c));
+		}
+	}
+	return auxiliaryColors;
+}
```

Comment:
I disagree with this approach; why store all the colors again? We have them in GameData already. Like I explained before, we just need their names. And then also the new `DisplayMessage::GetColor` can be removed.

Reply:
The problem is that on https://github.com/gemrb/gemrb/blob/5834c9ada3b08f2c77604351f20e8a82b4b76bed/gemrb/core/GameData.cpp#L640 the colors created is ` std::map<std::string, Color> colors;` so to map my enum I pass it as vector Color. 
The problem with this approach is that I have 13 colors defined on colors.2da so I have to map the positions to the enum.
I don't like this at all, for me will be better to map the enum to the names so there is no mistake there.